src/NativeScript/JSErrors.mm

```diff
@@ -25,12 +25,23 @@ void TNSSetUncaughtErrorHandler(TNSUncaughtErrorHandler handler) {
 
 void reportFatalErrorBeforeShutdown(ExecState* execState, Exception* exception) {
     GlobalObject* globalObject = static_cast<GlobalObject*>(execState->lexicalGlobalObject());
-    globalObject->inspectorController().reportAPIException(execState, exception);
+
+    JSValue uncaughtError = globalObject->get(execState, Identifier::fromString(execState, "__onUncaughtError"));
+    if (uncaughtError.isFunction()) {
+        CallData uncaughtErrorFunctionCallData;
+        CallType uncaughtErrorFunctionCallType = getCallData(uncaughtError, uncaughtErrorFunctionCallData);
+
+        MarkedArgumentBuffer uncaughtErrorArguments;
+        uncaughtErrorArguments.append(exception->value());
+        call(execState, uncaughtError, uncaughtErrorFunctionCallType, uncaughtErrorFunctionCallData, jsUndefined(), uncaughtErrorArguments);
```

Comment:
Use the overload of `call` that excepts an error out parameter - right now if the uncaught error handler throws an error itself, this will reset the VM's exception field, which we will have cleared previously.